tests/test_step.py

```diff
@@ -284,9 +284,9 @@ def test_step_list_args(config_file_list_arg_step):
     correctly.
     """
     config, returned_config_file = ListArgStep.build_config(
-        "science.fits", config_file=config_file_list_arg_step
+        "science.fits", config_file=str(config_file_list_arg_step)
```

Comment:
As updating `build_config` to also accept `Path` seems to match the efforts in this PR would you update that portion of `Step` to also accept a `Path` instance:
https://github.com/spacetelescope/stpipe/blob/ee5f1ee8e505b43faaad9e86409391274407d8b5/src/stpipe/step.py#L1376-L1381
That should make this call to `str` unnecessary.

Reply:
This actually fails within `ConfigObj` unfortunately.  It does not fully accept `pathlib.Path` objects.  And since this is now a dependency instead of vendored in `extern`, not much I can do here.  Though perhaps I could modify line 1379 to pass a `str()` of the path?

Reply:
It looks like `config_file` is returned by this function and is documented to be a `str`. So perhaps converting `config_file` to a `str` on 1377?